src/main/kotlin/study/racingcar/step2/Calculator.kt

```diff
@@ -0,0 +1,58 @@
+package study.racingcar.step2
+
+import study.racingcar.step2.operation.AbstractCalculation
+import study.racingcar.step2.operation.DivisionCalculation
+import study.racingcar.step2.operation.MinusCalculation
+import study.racingcar.step2.operation.MultiplicationCalculation
+import study.racingcar.step2.operation.PlusCalculation
+import java.util.HashMap
+
+class Calculator {
+    private val COMMON_SPACE = " "
+    private val calculationMap: MutableMap<String, AbstractCalculation>
+    private var expression: String? = null
+    private lateinit var expressionValues: Array<String>
+
+    init {
+        calculationMap = HashMap()
+        calculationMap["+"] = PlusCalculation()
+        calculationMap["-"] = MinusCalculation()
+        calculationMap["/"] = DivisionCalculation()
+        calculationMap["*"] = MultiplicationCalculation()
+    }
+
+    fun expression(expression: String): Calculator {
+        this.expression = expression
+        checkEmptyExpression()
+        return this
+    }
+
+    fun calculation(): Int {
+        checkEmptyExpression()
+        expressionValues = expression!!.split(COMMON_SPACE.toRegex()).toTypedArray()
+        return calculation(0)
+    }
+
+    private fun checkEmptyExpression() {
+        require(!(expression == null || expression!!.length == 0)) {
```

Comment:
length == 0이 아닌 아래와 같이 변경하면 어떨까요?

```suggestion
        require(!(expression == null || expression!!.isEmpty())) {
```

Reply:
추가로 가급적이면 nullable 타입의 사용을 지양하는게 좋을 것 같습니다.

아래와 같이 생성자 및 non null 타입의 사용을 고려해보세요

```kotlin
class Calculator(expression: String)

```